### PR TITLE
crypto: drivers: se050: ecc shared secret

### PR DESCRIPTION
**Pull request by ldts**
Allow clients to inject their own keypairs to derive the secret - the
previous implementation only allowed for secure element NVM based
keypairs to be used.

By default, the secure element does not store all the possible EC
curves in its internal memory; however attempting to inject a keypair
when the curve is not in the secure element would cause the injection
to fail.

This commit addresses that situation by generating those curves in the
SE whenever they are not available.

Tested with:
  TEE_ALG_ECDH_P192, 
  TEE_ALG_ECDH_P224, 
  TEE_ALG_ECDH_P256
  TEE_ALG_ECDH_P384 
   TEE_ALG_ECDH_P521 

**fixes xtest 4009**

Signed-off-by: Jorge Ramirez-Ortiz <jorge@foundries.io>

### CODE REVIEW

File: core/drivers/crypto/se050/core/ecc.c [hunk 1]
```diff
@@ -212,6 +212,9 @@ static TEE_Result se050_inject_public_key(sss_se05x_object_t *k_object,
 	if (st != kStatus_SSS_Success)
 		return TEE_ERROR_BAD_PARAMETERS;
 
+	(void)sss_se05x_key_store_create_curve(&se050_session->s_ctx,
```
Comment (jenswi-linaro) [inline core/drivers/crypto/se050/core/ecc.c:215]: What happens if this function fails?

Comment (ldts) [reply core/drivers/crypto/se050/core/ecc.c:215]: if the curve doesnt exist in the SE, the next function will fail to inject the key.
if the curve already exist in the SE and this function for whatever reason fails, it all will work as it should.

Apparently some versions of the API used to return an error if the curve was already in the SE (which would be a pass for what we need here) hence why I coded it this way. Not a big deal I dont think.

Comment (jenswi-linaro) [reply core/drivers/crypto/se050/core/ecc.c:215]: Makes sense. Please drop the `(void)` and add a comment along what you just said.

Comment (ldts) [reply core/drivers/crypto/se050/core/ecc.c]: @jenswi-linaro pushed

**Review by jenswi-linaro — COMMENTED** [review]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by ldts** [comment]
@jenswi-linaro added tag

**Pull request merged**